### PR TITLE
[JENKINS-50196] Fix nested Pod Templates support

### PR DESCRIPTION
**Pull request by Vlatombe**
Instead of using PodTemplateAction and NamespaceAction, pass a new
PodTemplateContext object that will be accessible by nested steps

### CODE REVIEW

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/NamespaceAction.java [hunk 1]
```diff
@@ -23,6 +27,7 @@ public NamespaceAction(Run run) {
         setRun(run);
     }
 
+    @Deprecated
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/NamespaceAction.java]: Redundant to deprecate members of a deprecated type.

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java [hunk 2]
```diff
@@ -99,6 +103,49 @@ public void runInPod() throws Exception {
...
+        p.setDefinition(new CpsFlowDefinition(loadPipelineScript(name.getMethodName() + ".groovy"), true));
+        WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+        assertNotNull(b);
+        List<PodTemplate> templates = cloud.getAllTemplates();
+
+        PodTemplate template = null;
+        while ((template = podTemplateWithLabel("mypod2", templates)) == null) {
+            LOGGER.log(Level.INFO, "Waiting for mypod2 template to be created");
+            templates = cloud.getAllTemplates();
+            Thread.sleep(1000);
```
Comment (jglick) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java]: BTW `SemaphoreStep` is usually helpful for things like this.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/NamespaceAction.java [hunk 3]
```diff
@@ -10,6 +10,10 @@
 import hudson.model.Run;
 import jenkins.model.RunAction2;
 
+/**
+ * Use <pre>getContext().get(PodTemplateContext.class)</pre> instead.
+ */
+@Deprecated
 public class NamespaceAction extends AbstractInvisibleRunAction2 implements RunAction2 {
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/NamespaceAction.java]: `AbstractInvisibleRunAction2` should also be deprecated, as you are deprecating both of its subtypes.

**Comment by fbelzunc** [comment]
@carlossg could you perform a release after this PR is merged?

**Pull request merged**